### DISCUSSION TITLE
Stop build from JS

### DISCUSSION THREAD

**Discussion by tsvecak** [vercel/next.js#28467] [Help] [answered]
Is there a way to stop a build purposefully? 
e.g. I check a boolean value on a fetched data and decide that if it's false I want to stop and fail the build?

**Comment by jamesmosier** [comment, marked as answer]
There is no internal method that Next exposes to stop a build. If you are fetching the data within `getStaticProps` you could throw an error: `new Error('something bad happened')` which would stop executing the buidl.
